packages/react/src/components/DetailsList/DetailsList.types.ts

```diff
@@ -436,6 +441,9 @@ export interface IColumn {
   /** Custom renderer for column header divider. */
   onRenderDivider?: IRenderFunction<IDetailsColumnProps>;
 
+  /** Custom renderer for filter icon. */
+  onRenderFilterIcon?: IRenderFunction<IDetailsColumnFilterIconProps>;
```

Comment:
Most commonly for icons, instead of a render function we'd just do `filterIconProps?: IIconProps`. But adding a render function is also fine if you prefer, or if you expect to need more customization in the future.

Reply:
Hey @ecraig12345, thanks for taking a look at this PR. I talked with Thomas and he preferred if we had full customization over the icon, in case we ever need further customizations in the future